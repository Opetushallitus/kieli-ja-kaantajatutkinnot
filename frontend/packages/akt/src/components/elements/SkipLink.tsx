import { Link } from '@mui/material';
import { FC } from 'react';

import { Text } from 'components/elements/Text';
import { Color } from 'enums/app';

export interface SkipLinkProps {
  text: string;
  href: string;
}

export const SkipLink: FC<SkipLinkProps> = ({ text, href }) => {
  return (
    <Link className="skip-link" href={href} color={Color.Secondary}>
      <Text>{text}</Text>
    </Link>
  );
};
