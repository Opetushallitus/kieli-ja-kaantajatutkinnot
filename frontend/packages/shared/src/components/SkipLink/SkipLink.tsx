import { Link } from '@mui/material';
import { FC } from 'react';

import { Text } from '../Text/Text';
import './SkipLink.scss';

export interface SkipLinkProps {
  text: string;
  href: string;
}

export const SkipLink: FC<SkipLinkProps> = ({ text, href }) => {
  return (
    <Link className="skip-link" href={href} color={'secondary'}>
      <Text>{text}</Text>
    </Link>
  );
};
