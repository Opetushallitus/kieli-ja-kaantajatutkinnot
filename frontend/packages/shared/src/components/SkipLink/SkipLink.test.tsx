import renderer from 'react-test-renderer';

import { SkipLink } from './SkipLink';

describe('SkipLink', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <SkipLink
          text="Link that looks like text"
          href="http://example.invalid"
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
