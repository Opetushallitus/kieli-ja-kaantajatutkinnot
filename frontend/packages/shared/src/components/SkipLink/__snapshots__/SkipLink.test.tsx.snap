// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink should render correctly 1`] = `
<a
  className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways skip-link css-tvmhkj-MuiTypography-root-MuiLink-root"
  href="http://example.invalid"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    Link that looks like text
  </p>
</a>
`;
